sdk/entra/WebJobs.Extensions.AuthenticationEvents/src/WebJobs.Extensions.AuthenticationEvents/AuthEventConfigProvider.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Azure.WebJobs.Description;
+using Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents.Framework;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Globalization;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web;
+
+namespace Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents
+{
+    /// <summary>The main configuration provider, this also handles the initial HTTP requests and response via IAsyncConverter.</summary>
+    /// <seealso cref="HttpRequestMessage" />
+    /// <seealso cref="HttpResponseMessage" />
+    [Extension("AuthenticationEvents", "AuthenticationEvents")]
+    internal class AuthEventConfigProvider : IExtensionConfigProvider, IAsyncConverter<HttpRequestMessage, HttpResponseMessage>
+    {
+        private readonly ILogger _logger;
+        private Uri _base_uri;
+
+        /// <summary>The listeners that are attached to the functions that implement the AuthenticationEventTriggerAttribute.</summary>
+        public Dictionary<string, AuthEventListener> Listeners { get; } = new Dictionary<string, AuthEventListener>();
+
+        /// <summary>Initializes a new instance of the <see cref="AuthEventConfigProvider" /> class.</summary>
+        /// <param name="loggerFactory">The logger factory from the WebJobs framework.</param>
+        public AuthEventConfigProvider(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger<AuthEventConfigProvider>();
+        }
+
+        /// <summary>Initializes the specified configuration.
+        /// This is where we create the main HTTP Listener endpoint and bind the trigger. (For all intents and purposes this is not obsolete).</summary>
+        /// <param name="context">The context we get from the Webjobs framework.</param>
+        [Obsolete("Is not obsolete marked by webjobs team, but chatted and this is correct. It is not being deprecated")]
+        public void Initialize(ExtensionConfigContext context)
+        {
+            context.AddBindingRule<AuthenticationEventsTriggerAttribute>()
+                    .BindToTrigger(new AuthEventBindingProvider(this));
+
+            _base_uri = context.GetWebhookHandler();
+            //LogInformation(string.Format(AuthEventResource.Log_EventHandler_Url, Uri));
+        }
+
+        internal void LogInformation(string message)
+        {
+            Console.WriteLine(message);
+            _logger.LogInformation(message);
+        }
+
+        internal void DisplayAzureFunctionInfoToConsole(string functionName)
```

Comment:
Can you explain why we are adding this? There are standard logs emitted by the functions host so I'm not sure this is necessary.